microsoft/testsuites/performance/storageperf.py

```diff
@@ -0,0 +1,165 @@
+# Licensed under the MIT license.
+from typing import Any, List
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    schema,
+    search_space,
+    simple_requirement,
+)
+from lisa.environment import Environment
+from lisa.features import Disk
+from lisa.notifier import DiskPerformanceMessage, DiskSetupType, DiskType
+from lisa.tools import FIOMODES, Fdisk, Fio, FIOResult, Lscpu, Mdadm
+from microsoft.testsuites.performance.common import handle_and_send_back_results
+
+
+def _format_disk(
+    node: Node,
+    disk_list: List[str],
+) -> List[str]:
+    fdisk = node.tools[Fdisk]
+    partition_disks: List[str] = []
+    for data_disk in disk_list:
+        fdisk.delete_partitions(data_disk)
+        partition_disks.append(fdisk.make_partition(data_disk, format=False))
+    return partition_disks
+
+
+def _stop_raid(node: Node) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.stop_raid()
+
+
+def _make_raid(node: Node, disk_list: List[str]) -> None:
+    mdadm = node.tools[Mdadm]
+    mdadm.create_raid(disk_list)
+
+
+@TestSuiteMetadata(
+    area="storage",
+    category="performance",
+    description="""
+    This test suite is to validate premium SSD data disks performance of Linux VM using
+     fio tool.
+    """,
+)
+class StoragePerformance(TestSuite):  # noqa
+    TIME_OUT = 3000
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance with 4K block size.
+        """,
+        priority=3,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(
+            min_core_count=72,
+            disk=schema.DiskOptionSettings(
+                disk_type=schema.DiskType.PremiumSSDLRS,
+                data_disk_iops=5000,
+                data_disk_count=search_space.IntRange(min=16),
+            ),
+        ),
+    )
+    def perf_premium_datadisks_4k(self, node: Node, environment: Environment) -> None:
+        self._perf_premium_datadisks(
+            node, environment, test_case_name="perf_premium_datadisks_4k"
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance using 1024K block size.
+        """,
+        priority=3,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(
+            min_core_count=72,
+            disk=schema.DiskOptionSettings(
+                disk_type=schema.DiskType.PremiumSSDLRS,
+                data_disk_iops=5000,
+                data_disk_count=search_space.IntRange(min=16),
+            ),
+        ),
+    )
+    def perf_premium_datadisks_1024k(
+        self, node: Node, environment: Environment
+    ) -> None:
+        self._perf_premium_datadisks(
+            node,
+            environment,
+            block_size=1024,
+            test_case_name="perf_premium_datadisks_4k",
+        )
+
+    def _perf_premium_datadisks(
```

Comment:
This logic looks similar with nvme. Considering to share more logic.

Reply:
working on it

Reply:
updated